src/main/kotlin/org/taktik/icure/asynclogic/samv2/impl/SamV2LogicImpl.kt

```diff
@@ -160,8 +166,17 @@ class SamV2LogicImpl(
 		return vmpDAO.findVmpsByGroupCode(vmpgCode, paginationOffset)
 	}
 
-	override fun findAmpsByLabel(language: String?, label: String?, paginationOffset: PaginationOffset<List<String>>): Flow<ViewQueryResultEvent> {
-		return ampDAO.findAmpsByLabel(language, label, paginationOffset)
+	override fun findAmpsByLabel(language: String?, label: String, paginationOffset: PaginationOffset<List<String>>): Flow<Amp> = flow {
+		val ampIds = ampDAO.listAmpIdsByLabel(language, label.split(" ").maxByOrNull { it.length }).toSet(TreeSet())
+		emitAll(
+			aggregateResults(
+				ids = ampIds,
+				limit = paginationOffset.limit,
+				supplier = { ids -> ampDAO.getEntities(ids) },
+				filter = { amp -> amp.officialName?.contains(label, true) ?: false },
```

Comment:
The requirement is to be able, thanks to this modification, to find all amps containing **all words of the label**. 
Example : 
If the user searches for "Dafalgan 50", he needs to find : 
- Dafalgan 50mg
- Dafalgan 500mg
- Dafalgan Efffervescent 500mg
==> The last one is not covered by your condition.